### ISSUE TITLE
Change default value for SPDLOG_BUILD_TESTING

### ISSUE THREAD

**Issue by taketwo** [gabime/spdlog#830]
Consider the following usage scenario: _spdlog_ is added as a Git submodule and then in the project CMakeLists included using: 

```cmake
add_subdirectory(third-party/spdlog)
```

This will result in _spdlog_ test targets being created because `SPDLOG_BUILD_TESTING` defaults to `ON`. One can force-reset this variable after the `add_subdirectory` call, but this is not convenient.

I propose to chose the default value based on whether _spdlog_ is used as a third-party dependency or as a standalone project. In other words, if _spdlog_ is included through `add_subdirectory` as in my scenario, the tests are disabled by default, and if _spdlog_ is configured standalone, then the tests are enabled by default (like they are now).

Some other projects follow this approach, for example [_pybind11_](https://github.com/pybind/pybind11/blob/ef13fb2e1c89f3e67f83eb860e8f6f37954a8cb1/CMakeLists.txt#L22-L29). Here is how they do this:

```cmake
# Check if pybind11 is being used directly or via add_subdirectory
set(PYBIND11_MASTER_PROJECT OFF)
if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
  set(PYBIND11_MASTER_PROJECT ON)
endif()


option(PYBIND11_INSTALL "Install pybind11 header files?" ${PYBIND11_MASTER_PROJECT})
option(PYBIND11_TEST    "Build pybind11 test suite?"     ${PYBIND11_MASTER_PROJECT})
```

I can submit a PR if you support such a change.

**Comment by gabime** [comment]
Sounds good. Thanks!

**Comment by taketwo** [comment]
I'm a bit confused. Against which branch do you want the pull request to be?

**Comment by gabime** [comment]
The v1.x branch please. The “master” is only for legacy

**Comment by gabime** [comment]
Fixed in #831 pr
